istioctl/pkg/util/formatting/formatter.go

```diff
@@ -62,9 +62,9 @@ func Print(ms diag.Messages, format string, colorize bool) (string, error) {
 }
 
 func printLog(ms diag.Messages, colorize bool) string {
-	var logOutput []string
-	for _, m := range ms {
-		logOutput = append(logOutput, render(m, colorize))
+	logOutput := make([]string, len(ms))
+	for i, m := range ms {
+		logOutput[i] = render(m, colorize)
```

Comment:
```suggestion
	logOutput := make([]string, 0, len(ms))
	for i, m := range ms {
		logOutput = append(logOutput, render(m, colorize))
```

Reply:
@zirain According to my tests, writing values using an index is more efficient than writing values using append when slice determines the length. Are there any other special cases?
<img width="832" alt="image" src="https://github.com/istio/istio/assets/47143494/e8139723-c595-42a1-843f-35135e88b160">

<img width="814" alt="image" src="https://github.com/istio/istio/assets/47143494/579c5752-7235-4503-b167-f7a46cf3f95f">

Reply:
> I am fine with pre-allocating, but please use `append`. Microbenchmarks on this nanosecond scale are not applicable to real world

@howardjohn Thanks~, I got it